### PR TITLE
DEVPROD-5149: Combine TSConfig

### PR DESCRIPTION
**Pull request by sophstad**
DEVPROD-5149

### Description
- Introduce shared `tsconfig.base.json` at root so that workspaces can extend it
- Upgrade `@types/node` since we were still using a version associated with Node 16

### CODE REVIEW

File: apps/spruce/tsconfig.json [hunk 1]
```diff
@@ -1,38 +1,8 @@
...
-    "jsxImportSource": "@emotion/react",
-    "types": [
-      "node",
-      "jest",
-      "@testing-library/jest-dom",
-      "new-relic-browser"
-    ],
     "baseUrl": "src",
-    "noFallthroughCasesInSwitch": true,
+    "strict": false
```
Comment (minnakt) [inline apps/spruce/tsconfig.json:5]: I wonder if we should turn this on! unless it causes a bunch of type errors 🫣 then maybe we should make a ticket

**Review by minnakt — APPROVED** [review]
yippeeeeeeee

File: apps/spruce/tsconfig.json (hunk 1 above)
Comment (sophstad) [reply apps/spruce/tsconfig.json:5]: Ya there are mad errors 😳 opened DEVPROD-5887!

File: tsconfig.base.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "forceConsistentCasingInFileNames": true,
+    "module": "CommonJS",
+    "moduleResolution": "Node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": true,
+    "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
```
[the remark this answers is not in the document]
Comment (sophstad) [reply tsconfig.base.json:20]: JSON does not support comments! ESLint was complaining so I removed them :)

**Pull request merged**